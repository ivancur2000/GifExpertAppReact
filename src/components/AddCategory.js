import React, {useState} from 'react';
import PropTypes  from 'prop-types';

const AddCategory = ({setCategories})=>{

  const [inputValue, setInputValue] = useState('');

  const handleOnChange = (e)=>{
    setInputValue(e.target.value);
  }

  const handleOnSubmit = (e)=>{
    e.preventDefault();
    if(inputValue.trim().length > 0){
      setCategories(category => [inputValue, ...category]);
      setInputValue('');
    }
  }

  return(
    <form onSubmit={handleOnSubmit}>
      <input 
        type="text"
        value={inputValue}
        onChange={handleOnChange}
      />
      <p >{inputValue}</p>
    </form>
  );
}
AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}
export default AddCategory;
