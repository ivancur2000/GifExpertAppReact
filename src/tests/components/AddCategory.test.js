import React from 'react';
import { shallow } from "enzyme"
import AddCategory from '../../components/AddCategory';


describe('Pruebas en el componente AddCategory', () => {
    
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories} />);
    beforeEach(() =>{
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    });

    test('debe de mostrarse correctamente AddCategory ', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de cambiar el input text', () => {
       const input = wrapper.find('input');
       const value = 'Coco';
       input.simulate('change', {target:{ value }});
       expect(wrapper.find('p').text().trim()).toBe(value);
    });
    
    test('no debe de postear la funcion onSubmit',()=>{
        wrapper.find('form').simulate('submit',{preventDefault(){}});
        expect(setCategories).not.toHaveBeenCalled();
    });

    test('debe de llamar el setCategories y limpiar la caja de texto', () => {
        const value = 'Spider';
        wrapper.find('input').simulate('change', { target: {value} })
        wrapper.find('form').simulate('submit', { preventDefault(){} })
        expect( setCategories ).toHaveBeenCalled();
        expect(wrapper.find('input').prop('value')).toBe('');
    });
    

})
