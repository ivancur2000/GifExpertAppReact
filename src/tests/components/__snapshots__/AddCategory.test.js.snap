// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente AddCategory debe de mostrarse correctamente AddCategory  1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <p />
</form>
`;
